addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/converter/MySensorsUpDownTypeConverter.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mysensors.converter;
+
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.mysensors.MySensorsBindingConstants;
+import org.openhab.binding.mysensors.internal.protocol.message.MySensorsMessage;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsVariable;
+
+/**
+ * Used to convert a String from an incoming MySensors message to a UpDownType
+ * 
+ * @author Andrea Cioni
+ *
+ */
+public class MySensorsUpDownTypeConverter implements MySensorsTypeConverter {
+
+    @Override
+    public Integer typeFromChannelCommand(String channel, Command command) {
+        if (channel.equals(MySensorsBindingConstants.CHANNEL_COVER)) {
+            if (command instanceof UpDownType) {
+                if (command.equals(UpDownType.UP)) {
+                    return MySensorsMessage.MYSENSORS_SUBTYPE_V_UP;
+                } else if (command.equals(UpDownType.DOWN)) {
+                    return MySensorsMessage.MYSENSORS_SUBTYPE_V_DOWN;
+                } else {
+                    throw new IllegalArgumentException("Invalid command of type UpDownType: " + command);
```

Comment:
You could throw all the IllegalArgumentException in the end of your method, it limits the possible paths to your code.